src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/runtime/Threads.java

```diff
@@ -155,7 +155,13 @@ private static synchronized void initialize(TypeDataBase db) {
         virtualConstructor.addMapping("NotificationThread", NotificationThread.class);
         virtualConstructor.addMapping("StringDedupThread", StringDedupThread.class);
         virtualConstructor.addMapping("AttachListenerThread", AttachListenerThread.class);
-        virtualConstructor.addMapping("DeoptimizeObjectsALotThread", DeoptimizeObjectsALotThread.class);
+
+        /* Only add DeoptimizeObjectsALotThread if it is actually present in the type database. */
+        try {
+            db.lookupType("DeoptimizeObjectsALotThread");
```

Comment:
I see there is `public Type lookupType(String cTypeName, boolean throwException);`, so this might be more accurately, without over-catching the exceptions:

```
  if (db.lookupType("DeoptimizeObjectsALotThread", false) != null) {
     virtualConstructor.addMapping("DeoptimizeObjectsALotThread", DeoptimizeObjectsALotThread.class);
  }
```

?

Reply:
Good idea. I've updated to do that.